lib/optimize/SideEffectsFlagPlugin.js

```diff
@@ -319,6 +322,12 @@ class SideEffectsFlagPlugin {
 		);
 	}
 
+	/**
+	 * @param {string} moduleName the module name
+	 * @param {undefined | boolean | string | string[]} flagValue the flag value
+	 * @param {Map<string, RegExp>} cache cache for glob to regexp
+	 * @returns {boolean | undefined} true, when the module has side effects, undefined or false when not
```

Comment:
Maybe I'm missing something, but when is the return value undefined? Every `switch` branch (for every input) returns a boolean.

<img width="883" alt="sideeffects-undefined" src="https://github.com/webpack/webpack/assets/3297808/60287d30-865d-4049-a3ba-90bb9764edd5">

TypeScript is fine with just `boolean`:

<img width="883" alt="sideeffects-boolean" src="https://github.com/webpack/webpack/assets/3297808/5f6c6274-a38a-49f4-9d5d-325126559979">

We are relying on webpack's handling of `sideEffects` in our projects, and it's important to get it right (and to adapt if anything changed).